iml-api/src/graphql.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::error::ImlApiError;
+use iml_postgres::{sqlx, PgPool};
+use juniper::{
+    http::{graphiql::graphiql_source, GraphQLRequest},
+    EmptyMutation, EmptySubscription, GraphQLEnum, RootNode,
+};
+use std::ops::Deref;
+use std::{convert::Infallible, sync::Arc};
+use warp::Filter;
+
+#[derive(juniper::GraphQLObject)]
+/// A Corosync Node found in `crm_mon`
+struct CorosyncNode {
+    /// The name of the node
+    name: String,
+    /// The id of the node as reported by `crm_mon`
+    id: String,
+    online: bool,
+    standby: bool,
+    standby_onfail: bool,
+    maintenance: bool,
+    pending: bool,
+    unclean: bool,
+    shutdown: bool,
+    expected_up: bool,
+    is_dc: bool,
+    resources_running: i32,
+    r#type: String,
+}
+
+pub(crate) struct QueryRoot;
+
+#[derive(GraphQLEnum)]
+enum SortDir {
+    Asc,
+    Desc,
+}
+
+impl Default for SortDir {
+    fn default() -> Self {
+        Self::Asc
+    }
+}
+
+impl Deref for SortDir {
+    type Target = str;
+
+    fn deref(&self) -> &Self::Target {
+        match self {
+            Self::Asc => "asc",
+            Self::Desc => "desc",
+        }
+    }
+}
+
+#[juniper::graphql_object(Context = Context)]
+impl QueryRoot {
+    #[graphql(arguments(
```

Comment:
This effectively gives a default it looks like. Seems like a good solution to having to explicitly specify a limit/direction on every call.

Reply:
This is just docs for the query browser. The default is due to the `unwrap_or` calls within this fn.